gcm_filters/kernels.py

```diff
@@ -51,6 +58,142 @@ def __call__(self, field: ArrayType):
 ALL_KERNELS[GridType.CARTESIAN] = CartesianLaplacian
 
 
+@dataclass
+class MOM5LaplacianU(BaseLaplacian):
+    """Laplacian for MOM5 gird (velocity points)"""
+
+    dxt: ArrayType
+    dyt: ArrayType
+    dxu: ArrayType
+    dyu: ArrayType
+    area_u: ArrayType
+    wet: ArrayType
+
+    def __call__(self, field: ArrayType):
+        """Uses code by Elizabeth"""
+        np = get_array_module()
+        field = np.nan_to_num(field)
+        fx = 2 * (np.roll(field, shift=-1, axis=0) - field)
+        fx /= np.roll(self.dxt, -1, axis=0) + np.roll(self.dxt, (-1, -1), axis=(0, 1))
+        fy = 2 * (np.roll(field, shift=-1, axis=1) - field)
+        fy /= np.roll(self.dyt, -1, axis=1) + np.roll(self.dyt, (-1, -1), axis=(0, 1))
+        out1 = 0.5 * fx * (self.dyu + np.roll(self.dyu, -1, axis=0))
+        out1 -= 0.5 * np.roll(fx, 1, axis=0) * (self.dyu + np.roll(self.dyu, 1, axis=0))
+        out1 /= self.area_u
+        out2 = 0.5 * fy * (self.dxu + np.roll(self.dxu, -1, axis=1))
+        out2 -= 0.5 * np.roll(fy, 1, axis=1) * (self.dxu + np.roll(self.dxu, 1, axis=1))
+        out2 /= self.area_u
+        return (out1 + out2) * self.wet
```

Comment:
To ensure a conservative Laplacian, I would multiply the wet mask at the flux level, i.e., when you compute the fluxes between neighboring grid cells in lines 80, 81, 83, 84. You could have a look at how I applied the wet mask in my `IrregularCartesianLaplacianWithLandMask` or `POPTripolarLaplacianTpoint` Laplacians (the latter is in PR#29). The `IrregularCartesianLaplacianWithLandMask` is just another version of your `MOM5LaplacianU` and `MOM5LaplacianT` Laplacians, but with different grid labeling (see the comments https://github.com/ocean-eddy-cpt/gcm-filters/issues/24#issuecomment-778903595 and https://github.com/ocean-eddy-cpt/gcm-filters/issues/24#issuecomment-779952899). So we should be able to get your conservation issues fixed by comparing notes.